src/dinitcheck.cc

```diff
@@ -765,7 +765,10 @@ service_record *load_service(service_set_t &services, const std::string &name,
 
     auto check_command = [&](const char *setting_name, const char *command) {
         struct stat command_stat;
-        if (fstatat(dirfd, command, &command_stat, 0) == -1) {
+        if (command[0] != '/') {
+            report_service_description_err(name,
+                    std::string("executable '") + command + "' is not an absolute path");
+        } else if (fstatat(dirfd, command, &command_stat, 0) == -1) {
```

Comment:
The standard formatting used in Dinit is to have the "else if" on the next line after the closing brace (see the CODE-STYLE document).

Using "else" here means that if the path is both non-absolute and refers to a non-existent (non-stat'able) file, only one warning will be printed. I think it should do both checks.

Reply:
Will fix the style.

> Using "else" here means that if the path is both non-absolute and refers to a non-existent (non-stat'able) file, only one warning will be printed. I think it should do both checks.

Yes, I skip them because if executable is not specified in absolute form, dinitcheck may end up examining wrong files and print meaningless warnings.

For example,

```
$ doas dinitcheck
...
Checking service: seatd...                                                                                                                                     
Service 'seatd': command executable 'seatd' is not executable by owner.
...
$ cat /usr/lib/dinit.d/seatd 
type               = process
command            = seatd -n 3 -g seat
smooth-recovery    = true
logfile            = /var/log/seatd.log
ready-notification = pipefd:3
depends-on         = rc.target
before             = login.target
```

dinitcheck stats `/lib/dinit.d/seatd` instead of `/usr/bin/seatd`, the diagnose is probably meaningless (and misleading) if we aren't sure what the executable is.

Reply:
Ok, that's a fair point.